File: hamming_distance/hamming_distance.py
# -*- coding: utf-8 -*-
"""
Created on Wed Jan 26 12:35:50 2022

@author: marcuslindskog
"""

class LengthError(Exception):
    pass

def hamming(strand1, strand2):
    hamming_dist = 0
    
    if len(strand1) != len(strand2):
        raise LengthError('Strands are not of the same length')
    
    for c in range(0,len(strand1)):
        if strand1[c] == strand2[c]:
            hamming_dist = hamming_dist + 1
            
    return(hamming_dist)
    
    
#Tests
y = "ABSD"
x = "ABSDD"

hamming(y, x)

y = "ABSDD"
x = "ABSDD"

hamming(y, x)

y = "CDSDA"
x = "ABSDD"

hamming(y, x)

File: hamming_distance/hamming_distance.R
hamming <- function(strand1, strand2){
  v_strand1 <- unlist(strsplit(strand1, split =""))
  v_strand2 <- unlist(strsplit(strand2, split =""))
  
  if(length(v_strand1) == length(v_strand2)){
    sum(v_strand1 != v_strand2)
  }else{
    stop("Strands are not of equal length")
  }
  
}

strand1 <- "CGC"
strand2 <- "CGTC"
q  <- hamming(strand1, strand2)
